### PR TITLE
cleanup: rustc doesn't use an external archiver

### PR DESCRIPTION
**Pull request by petrochenkov**
cc https://github.com/rust-lang/rust/pull/45090
r? @alexcrichton

### CODE REVIEW

File: config.toml.example [hunk 1]
```diff
@@ -250,14 +250,11 @@
 # Whether or not `panic!`s generate backtraces (RUST_BACKTRACE)
 #backtrace = true
 
-# The default linker that will be used by the generated compiler. Note that this
-# is not the linker used to link said compiler.
+# The default linker that will be hard-coded into the generated compiler for
+# targets that don't specify linker explicitly in their target specifications.
+# Note that this is not the linker used to link said compiler.
```
Comment (petrochenkov) [inline config.toml.example]: A drive-by comment improvement.

**Comment by alexcrichton** [comment]
r=me

Thanks!

**Comment by kennytm** [comment]
@bors retry 

* Android SDK HTTPS issue (fixed in #45193)
